front/src/modules/ui/table/options/components/TableOptionsDropdownContent.tsx

```diff
@@ -0,0 +1,253 @@
+  const [viewEditMode, setViewEditMode] = useRecoilState(
+    tableViewEditModeState,
+  );
+  const [views, setViews] = useRecoilScopedState(
+    tableViewsState,
+    TableRecoilScopeContext,
+  );
+  const visibleColumns = useRecoilValue(visibleTableColumnsState);
+  const hiddenColumns = useRecoilValue(hiddenTableColumnsState);
+  const viewsById = useRecoilScopedValue(
+    tableViewsByIdState,
+    TableRecoilScopeContext,
+  );
+
+  const handleColumnVisibilityChange = useCallback(
+    (columnId: string, nextIsVisible: boolean) => {
+      const nextColumns = columns.map((column) =>
+        column.id === columnId
+          ? { ...column, isVisible: nextIsVisible }
+          : column,
+      );
+
+      (onColumnsChange ?? setColumns)(nextColumns);
+    },
+    [columns, onColumnsChange, setColumns],
+  );
+
+  const renderFieldActions = useCallback(
+    (column: ViewFieldDefinition<ViewFieldMetadata>) =>
+      // Do not allow hiding last visible column
+      !column.isVisible || visibleColumns.length > 1 ? (
+        <IconButton
+          icon={
+            column.isVisible ? (
+              <IconMinus size={theme.icon.size.sm} />
+            ) : (
+              <IconPlus size={theme.icon.size.sm} />
+            )
+          }
+          onClick={() =>
+            handleColumnVisibilityChange(column.id, !column.isVisible)
+          }
+        />
+      ) : undefined,
+    [handleColumnVisibilityChange, theme.icon.size.sm, visibleColumns.length],
+  );
+
+  const resetViewEditMode = useCallback(() => {
+    setViewEditMode({ mode: undefined, viewId: undefined });
+
+    if (viewEditInputRef.current) {
+      viewEditInputRef.current.value = '';
+    }
+  }, [setViewEditMode]);
+
+  const handleViewNameSubmit = useCallback(
+    (event?: FormEvent) => {
+      event?.preventDefault();
+
+      if (viewEditMode.mode && viewEditInputRef.current?.value) {
+        const name = viewEditInputRef.current.value;
+        const nextViews =
+          viewEditMode.mode === 'create'
+            ? [...views, { id: v4(), name }]
+            : views.map((view) =>
+                view.id === viewEditMode.viewId ? { ...view, name } : view,
+              );
+
+        (onViewsChange ?? setViews)(nextViews);
+      }
+
+      resetViewEditMode();
+    },
+    [
+      onViewsChange,
+      resetViewEditMode,
+      setViews,
+      viewEditMode.mode,
+      viewEditMode.viewId,
+      views,
+    ],
+  );
+
+  const handleSelectOption = useCallback(
+    (option: Option) => {
+      handleViewNameSubmit();
+      setSelectedOption(option);
+    },
+    [handleViewNameSubmit],
+  );
+
+  const resetSelectedOption = useCallback(() => {
+    setSelectedOption(undefined);
+  }, []);
+
+  useEffect(() => {
+    if (!isDropdownButtonOpen) {
+      handleViewNameSubmit();
+      resetSelectedOption();
+    }
+  }, [handleViewNameSubmit, resetSelectedOption, isDropdownButtonOpen]);
+
+  useScopedHotkeys(
+    Key.Escape,
```

Comment:
`Key.Enter` is missing here. In the previous behavior, pressing the `Enter` key submitted the `viewName` input + closed the dropdown.